sdk/core/azure-core/src/http/user_agent.cpp

```diff
@@ -7,13 +7,12 @@
  */
```

Comment:
Consider moving the contents of this file to `http.cpp`, so it lives with the rest of the HttpShared implementation.

Reply:
User agent logic is pretty complicated, I think it is better for it to live in the `user_agent.cpp`.

Reply:
It seems odd to have a source file name, which doesn't correspond to any namespace or class name. Hard to discover.

`http.cpp` already contains the implementation of `InsertHeaderWithValidation` and constants from HttpShared (and only that), it seems like a perfect place for another method on the same type, in the same header rather than creating a separate file for another method.

With our current layout in this PR, we have two methods on `HttpShared`, one implemented in `http.hpp`, and the other in a separate source file than where the constants are defined. Seems odd.
https://github.com/Azure/azure-sdk-for-cpp/blob/f121f81fdca8cb3aac1fd09eda76f52f4d17861f/sdk/core/azure-core/src/http/http.cpp#L17-L21

That said, it's implementation detail and we can change later.
Up to you to make the call.

Reply:
I can see reasons for both, I talked to Larry, he prefers this. As you said, it is an implementation detail that we can change later. Let's keep it here for now for a smaller diff when things are going into release, less chance to have something wrong and easier to follow/review.

Reply:
> Let's keep it here for now for a smaller diff when things are going into release, less chance to have something wrong and easier to follow/review.

That's a good reason. Thanks!

Reply:
The user agent logic is *COMPLICATED* - it involves registry queries, and carries a TON of baggage along with it.

I don't want to complicate http.cpp (which is mostly just text constants) with that code, and there is nothing in our guidelines or our practices which *require* that all the methods in a class implementation be held in an implementation.cpp file (simple example: WinHttpRequest.hpp has most of its implementation code in WinHttpTransport.cpp).

Moving this into http.cpp is churn for the sake of churn.